llvm/lib/Target/AArch64/GISel/AArch64RegisterBankInfo.h

```diff
@@ -139,6 +139,10 @@ class AArch64RegisterBankInfo final : public AArch64GenRegisterBankInfo {
   bool onlyDefinesFP(const MachineInstr &MI, const MachineRegisterInfo &MRI,
                      const TargetRegisterInfo &TRI, unsigned Depth = 0) const;
 
+  /// \returns true if \p MI can take both fpr and gpr uses, but prefers fp.
+  bool prefersFPUse(const MachineInstr &MI, const MachineRegisterInfo &MRI,
+                    const TargetRegisterInfo &TRI, unsigned Depth = 0) const;
```

Comment:
```suggestion
                    const AArch64RegisterInfo &TRI, unsigned Depth = 0) const;
```
TRI should probably just be a field in the RegisterBankInfo